reco_analysis/reco_analysis/chatbot_app/streamlit_app.py

```diff
@@ -218,27 +225,31 @@ def handle_user_input(agent, prompt: str | None):
         with st.chat_message("Patient", avatar=get_icon("Patient")):
             st.markdown(prompt)
         st.session_state.turn = "Doctor"  # Next turn is for the doctor
-        
+
         # Generate and display doctor's response (including adding to memory)
         response = agent.generate_response()  # Generate the response
         stream_response("Doctor", response)
         st.session_state.turn = "Patient"  # Next turn is for the patient
 
+
 def end_conversation(agent: DialogueAgent, session):
     """
     Ends the conversation and marks the session as completed.
 
     Args:
         agent (DialogueAgent): The dialogue agent managing the conversation.
         session (Session): The SQLAlchemy session object.
-    """    
+    """
     st.session_state.conversation_ended = True
     agent.end_conversation = True
-    convo_session: data_models.ConversationSession = (
-        data_models.ConversationSession.get_by_id(agent.session_id, session)
+    convo_session: data_models.ConversationSession = data_models.ConversationSession.get_by_id(
+        agent.session_id, session
     )
     convo_session.mark_as_completed(session)
 
+    # also generate a summary email
+    summarizer_job.summarize_conversation(conversation_session_id=convo_session.id)
```

Comment:
this is the only change/addition. everything else in this file is an auto-formatting change.